libr/util/idpool.c

```diff
@@ -31,10 +48,12 @@ R_API bool r_id_pool_grab_id(RIDPool* pool, ut32* grabber) {
 		return false;
 	}
 	if (pool->freed_ids) {
-		ut32 grab = (ut32) (size_t)r_queue_dequeue (pool->freed_ids);
-		*grabber = (ut32) grab;
-		if (r_queue_is_empty (pool->freed_ids)) {
-			r_queue_free (pool->freed_ids);
+		RBTreeIter iter = r_rbtree_first (pool->freed_ids);
+		*grabber = (ut32)(size_t)(iter.path[iter.len-1]->data);
+//		eprintf ("deleting %d from the tree\n", *grabber);
+		r_rbtree_delete (pool->freed_ids, (void *)(size_t)(*grabber), NULL);
+		if (r_rbtree_size (pool->freed_ids) == 1) {
```

Comment:
I think there is no need to free the tree. This simplifies the logic here and in the _delete function.